### DISCUSSION TITLE
773

### DISCUSSION THREAD

**Discussion by giscus** [jojoldu/blog-comments#946] [Comments]
# 773

최근에 Pete Hodgson가 martinfowler 블로그에 기재한 글을 보면서 로깅도 하나의 기능으로 봐야한다는 생각이 더 강해져서 이 글을 쓰게 되었다. 시스템을 구축하다보면 다음과 같이 크게 두 종류의 로그를 남긴다. logger.error(`API Timeout: ${timeout} seconds`, e); logger.info(`Ordered Food Product Id: ${product.id}`); logger.debug(`Total Order Price Amount: ${sum(products.amount)}`); 이 로그들은 모두 필요한 메세지를 노출시키지만, 실상은 목적이 다르다. Error와 Info 로그는 운영 환경에서 장애를 디버깅하거나 실행 중인 시스템의 여러 진행 상황, 지..

https://jojoldu.tistory.com/773

**Comment by hyunha95** [comment]
좋은 글 감사합니다. 회사에서 개발 환경과 운영 환경의 로그를 다르게 출력하는 것을 보고 토이 프로젝트에도 비슷하게 도입했습니다. 예를 들어 운영 환경에서는 sql관련 로그를 출력하지 않고 개발환경에서는 sql관련 로그를 출력하는 것 등이 있었습니다. 그러면서도 운영환경에서 어디까지 로그를 남겨야하는지 계속 고민이었고 또한 그러한 로그들이 비즈니스 로직에 너무 많이 영향을 준다는게 항상 고민이었습니다. 

SI에서 일을 하다보니 솔직히 익셉션 로그를 제외한 로그의 실익을 체감하지 못하고 있습니다. 현재 개발하고 있는 토이 프로젝트가 잘되어서 운영 경험을 해볼 수 있게 된다면 이 글을 다시 읽고 지금 당장 이해하지 못했던 부분을 이해할 수 있게 되는 시간이 왔으면 좋겠습니다
